hail_batch_dev.md

```diff
@@ -0,0 +1,133 @@
+# Hail Batch developer setup
+
+If you only intend to use Hail Batch for running your pipelines, you can skip this
+document. This is meant for developers that work on the Hail Batch codebase.
+
+1. Instead of using the standard sign-up link, ask another existing Hail developer to create a Hail developer account for you. If you've already signed up, that's okay too, but [a little more work to fix](https://github.com/hail-is/hail/blob/main/dev-docs/creating-a-developer-account.md).
+1. Ask to be added to the `hail-dev@populationgenomics.org.au` permissions group.
+1. Follow the instructions to add an [OAuth 2.0 redirect URI](https://github.com/hail-is/hail/blob/main/dev-docs/creating-a-developer-account.md), but note that our GCP project is called `hail-295901`. If your email address is `jane.doe@populationgenomics.org.au`, your Hail `$USERNAME` will be `janedoe` (i.e. does not contain a dot).
+1. Install Hail:
+
+   ```bash
+   conda create --name hail -c cpg -c bioconda -c conda-forge hail
+   conda activate hail
+   ```
+
+1. Retrieve a new Hail token:
+
+   ```bash
+   hailctl auth login
+   ```
+
+1. Prepare a change and push it to `$BRANCH` in our Hail fork. Then run a `dev deploy`, which will run the steps you specify as defined in the `build.yaml` file in your branch. The corresponding Kubernetes deployments will be brought up in your own separate namespace, so they won't interfere with the production deployment.
+
+   ```bash
+   hailctl dev deploy --branch populationgenomics/hail:$BRANCH --steps deploy_batch deploy_query
+   ```
+
+1. The previous step should have printed a link to a CI dashboard. Follow the progress in the CI dashboard and wait until all steps have succeeded.
+1. The first time you run a `dev deploy`, you'll also have to create a new user. Unless you delete the "auth tables" in your namespace, subsequent deploys will be able to see the same user. At the moment, this initial setup requires a few manual steps. Steps below assume that you've set up the environment `$USERNAME` accordingly, e.g.
+
+   ```bash
+   export USERNAME=janedoe
```

Comment:
On my laptop, this export doesn't work, probably because clashes with the system environment variable `$USERNAME`:

```
$ echo $USERNAME
vlad
$ export USERNAME=vladislavsavelyev
$ echo $USERNAME
vlad
```

Perhaps could use `$NAMESPACE` instead as the variable name in the docs?

Reply:
Argh, that's good to know. I've used `$USERNAME` because that's also used in the Hail docs. Renamed it to $NAMESPACE.